src/network/grpc/client.rs

```diff
@@ -67,3 +71,38 @@ fn subscribe(
             Ok((node_id, prop_handles))
         })
 }
+
+// Fetches a block from a network peer in a one-off, blocking call.
+// This function is used during node bootstrap to fetch the genesis block.
+pub fn fetch_block(peer: Peer, hash: &HeaderHash) -> Result<Block, ()> {
```

Comment:
I'd rather like this function to return a proper error code.

You can now create new error type fairly easily:
```rust
custom_error! {pub FetchBlockError
    // ...
}